services/MerchStoreService.ts

```diff
@@ -107,47 +107,69 @@ export default class MerchStoreService {
       const merchItem = MerchandiseItemModel.create({ ...item, collection });
       await merchItemRepository.upsertMerchItem(merchItem);
       return merchItemRepository.findByUuid(merchItem.uuid);
     });
   }
 
+  /**
+   * Verify that items have valid options. An item with variants disabled cannot have multiple
+   * options, and an item with variants enabled cannot have multiple option types.
+   */
+  private static verifyItemHasValidOptions(item: MerchItem | MerchandiseItemModel) {
+    if (!item.hasVariantsEnabled && item.options.length > 1) {
+      throw new UserError('Merch items with variants disabled cannot have multiple options');
+    }
+    if (item.hasVariantsEnabled && MerchStoreService.hasMultipleOptionTypes(item.options)) {
+      throw new UserError('Merch items cannot have multiple option types');
+    }
+  }
+
+  private static hasMultipleOptionTypes(options: MerchItemOption[]) {
+    const optionTypes = new Set(options.map((option) => option.metadata.type));
+    return optionTypes.size > 1;
+  }
+
   public async editItem(uuid: Uuid, itemEdit: MerchItemEdit): Promise<MerchandiseItemModel> {
     return this.transactions.readWrite(async (txn) => {
       const merchItemRepository = Repositories.merchStoreItem(txn);
       const item = await merchItemRepository.findByUuid(uuid);
       if (!item) throw new NotFoundError();
       const { options, collection: updatedCollection, ...changes } = itemEdit;
 
-      const merchItemOptionRepository = Repositories.merchStoreItemOption(txn);
-      const updatedOptions = await Promise.all(options.map(async (optionUpdate) => {
-        const option = await merchItemOptionRepository.findByUuid(optionUpdate.uuid);
-        if (!option) throw new NotFoundError('Item option not found');
-        // 'quantity' is incremented instead of directly set to avoid concurrency issues with orders
-        // e.g. there's 10 of an item and someone adds 5 to stock while someone else orders 1
-        // so the merch store admin sets quantity to 15 but the true quantity is 14
-        if (optionUpdate.quantity) optionUpdate.quantity += option.quantity;
-        return merchItemOptionRepository.upsertMerchItemOption(option, optionUpdate);
-      }));
+      if (options) {
+        const updatedOptionsByUuid = new Map(options.map((option) => [option.uuid, option]));
+        item.options.map((option) => {
+          if (!updatedOptionsByUuid.has(option.uuid)) return;
+          const updatedOption = updatedOptionsByUuid.get(option.uuid);
+          // 'quantity' is incremented instead of directly set to avoid concurrency issues with orders
+          // e.g. there's 10 of an item and someone adds 5 to stock while someone else orders 1
+          // so the merch store admin sets quantity to 15 but the true quantity is 14
+          if (updatedOption.addQuantity) updatedOption.addQuantity += option.quantity;
+          return MerchandiseItemOptionModel.merge(option, updatedOption);
```

Comment:
Does this work? We assign the new quantity to `updatedOption.addQuantity` and then merge the option and its update, so I think `addQuantity` would be ignored by `MerchandiseItemOptionModel` since the model doesn't have an `addQuantity` field.

Maybe instead we do something like
```suggestion
          updatedOption.addQuantity ??= 0; // or move this line to the controller?
          return MerchandiseItemOptionModel.merge(option, {
            ...updatedOption,
            quantity: option.quantity + updatedOption.addQuantity,
          });
```

Reply:
You're right that it doesn't currently work, but isn't the fix to just reverse the assignment? i.e.
```ts
if (updatedOption.addQuantity) option.quantity+= updatedOption.addQuantity;
```
e.g. let's say there's 10 of an option in stock, and we add 5 more to it's stock, then `addQuantity` is 5 and and `quantity` is 10 so we set `quantity = 15` instead of `addQuantity = 15`.

I think the variable naming is slightly confusing currently since `updatedOption` isn't really an updated option, it just a partial object that contains some of the updates that need to be applied to the currently existing option. `option` is the object we need to update, and we should make that clearer in it's naming, so I'll rename the variables accordingly

Reply:
That works too, but I didn't want to edit the original `option` object so I moved that operation inside the model merge.